### PR TITLE
HTML-CSS-Popup 

### PR DESCRIPTION
**Pull request by kovalov**
### HTML-CSS-Popup 

[Demo](https://kovalov.github.io/dropdown-menu-css/)
[Code base](https://github.com/kovalov/dropdown-menu-css)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

**Comment by artem-trubin** [comment]
Hi! Nice usage of sprite. 
Though, there is a problem. You cannon open the popup menu with keyboard. And there is no visual identification when focusing "More" button.

File: submissions/kovalov/html-css-popup/index.html [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta
+      name="viewport"
+      content="width=device-width, initial-scale=1.0"
+    />
+    <link rel="stylesheet" href="styles.css" />
+    <title>Document</title>
```
Comment (A-Ostrovnyy) [inline submissions/kovalov/html-css-popup/index.html:11]: Add corresponding page title

File: submissions/kovalov/html-css-popup/styles.css [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+	box-sizing: border-box;
+	padding: 0;
+	margin: 0;
+}
+
+body {
+	font-family: Arial, Helvetica, sans-serif;
+}
+
+.header {
```
Comment (A-Ostrovnyy) [inline submissions/kovalov/html-css-popup/styles.css:13]: Avoid empty rules, just remove them)

File: submissions/kovalov/html-css-popup/styles.css [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+}
+
+.header__container {
+	display: flex;
+	align-items: center;
+	justify-content: flex-end;
+	padding: 0.5rem;
+}
+
+.navigation {}
```
Comment (A-Ostrovnyy) [inline submissions/kovalov/html-css-popup/styles.css:23]: Avoid empty rules, just remove them)

File: submissions/kovalov/html-css-popup/styles.css [hunk 4]
```diff
@@ -0,0 +1,232 @@
+*,
```
Comment (A-Ostrovnyy) [inline submissions/kovalov/html-css-popup/styles.css:1]: :fire:

File: submissions/kovalov/html-css-popup/styles.css [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+*::before,
+*::after {
+	box-sizing: border-box;
+	padding: 0;
+	margin: 0;
+}
+
+body {
+	font-family: Arial, Helvetica, sans-serif;
```
Comment (A-Ostrovnyy) [inline submissions/kovalov/html-css-popup/styles.css:10]: :fire:

File: submissions/kovalov/html-css-popup/styles.css [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+	border-radius: 50%;
+	transition: opacity 0.25s ease-in-out;
+}
+
+.dropdown:hover.dropdown::before {
+	opacity: 1;
+}
+
+.dropdown__control {
+	appearance: none;
```
Comment (A-Ostrovnyy) [inline submissions/kovalov/html-css-popup/styles.css:71]: :fire:

File: submissions/kovalov/html-css-popup/styles.css [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+
+.button:last-child {
+	margin-right: 0;
+}
+
+.button__icon {
+	height: 32px;
+	width: 32px;
+	color: #333;
+}
```
Comment (A-Ostrovnyy) [inline submissions/kovalov/html-css-popup/styles.css:232]: Avoid empty lines at the end of files. [Reason](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline?noredirect=1&lq=1)

File: submissions/kovalov/html-css-popup/index.html [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+            tabindex="-1"
+          />
+          <label
+            class="dropdown__toggler"
+            for="dropdown-control-1"
+            id="dropdown-toggler-1"
+            tabindex="0"
+          >
+            <svg class="dropdown__toggler-icon">
+              <use href="sprite.svg#menu-icon"></use>
```
Comment (A-Ostrovnyy) [inline submissions/kovalov/html-css-popup/index.html:40]: :fire:

File: submissions/kovalov/html-css-popup/index.html [hunk 9]
```diff
@@ -0,0 +1,211 @@
...
+          <ul class="navigation__list">
+            <li class="navigation__list-item">
+              <a href="#" class="navigation__list-link"> Mail </a>
+            </li>
+            <li class="navigation__list-item">
+              <a href="#" class="navigation__list-link"> Images </a>
+            </li>
+          </ul>
+        </nav>
+        <div class="dropdown">
```
Comment (A-Ostrovnyy) [inline submissions/kovalov/html-css-popup/index.html:26]: Why did you remove all items with icons(no text) from the navigation?

File: submissions/kovalov/html-css-popup/index.html [hunk 10]
```diff
@@ -0,0 +1,211 @@
...
+          <input
+            class="dropdown__control"
+            type="checkbox"
+            id="dropdown-control-1"
+            tabindex="-1"
```
Comment (A-Ostrovnyy) [inline submissions/kovalov/html-css-popup/index.html:31]: This attribute broke the native browser order of interactive elements, use it wisely.

File: submissions/kovalov/html-css-popup/index.html [hunk 11]
```diff
@@ -0,0 +1,211 @@
...
+                  class="inner-dropdown__control"
+                  type="checkbox"
+                  id="dropdown-control-2"
+                />
+                <label
+                  class="inner-dropdown__toggler"
+                  for="dropdown-control-2"
+                  id="dropdown-toggler-2"
+                >
+                  More
```
Comment (A-Ostrovnyy) [inline submissions/kovalov/html-css-popup/index.html:148]: Fix styles for focus and hovered states

image

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Good work!

There is no need to add images or other graphics to this or future PR's.

**Comment by kovalov** [comment]
Hi, everyone!
I've amended project files.
Please review and comment.

File: submissions/kovalov/html-css-popup/index.html [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+                <input
+                  class="dropdown__control"
+                  type="checkbox"
+                  id="dropdown-1"
+                />
+                <div class="dropdown__content">
+                  <ul class="dropdown__list">
+                    <li class="dropdown__list-item">
+                      <a href="#" class="dropdown__link">
+								<span class="dropdown__link-title">Item One</span>
```
Comment (artem-trubin) [inline submissions/kovalov/html-css-popup/index.html:39]: These are some wild indentations, something went out of control here?)

**Review by artem-trubin — COMMENTED** [review]
Nice custom design!

File: submissions/kovalov/html-css-popup/README.md [hunk 13]
```diff
@@ -0,0 +1,3 @@
+# dropdown-menu-css
```
Comment (A-Ostrovnyy) [inline submissions/kovalov/html-css-popup/README.md:1]: Remove files not for review: images, compiled files, configs, add only source code

File: submissions/kovalov/html-css-popup/styles.css [hunk 14]
```diff
@@ -0,0 +1,219 @@
+:root {
```
Comment (A-Ostrovnyy) [inline submissions/kovalov/html-css-popup/styles.css:1]: :fire:

File: submissions/kovalov/html-css-popup/styles.css [hunk 15]
```diff
@@ -0,0 +1,219 @@
...
+.sub-dropdown__control:checked ~ .sub-dropdown__content {
+	display: block;
+}
+
+.sub-dropdown__control:checked ~ .sub-dropdown__toggler {
+	display: none;
+}
+
+.sub-dropdown__control:checked {
+	display: none;
```
Comment (A-Ostrovnyy) [inline submissions/kovalov/html-css-popup/styles.css:218]: Why do you use `display: none` if previously you used `appearance: none`?

File: submissions/kovalov/html-css-popup/index.html [hunk 16]
```diff
@@ -0,0 +1,112 @@
...
+    <link rel="stylesheet" href="styles.css" />
+    <title>Kottans | Popup</title>
+  </head>
+  <body>
+    <header class="header">
+      <div class="header__container">
+        <nav class="navigation">
+          <ul class="navigation__list">
+            <li class="navigation__list-item">
+              <a href="#" class="header__title navigation__link">
```
Comment (A-Ostrovnyy) [inline submissions/kovalov/html-css-popup/index.html:19]: Logically this element looks like a page logo and not like part of page navigation, if you want to add a logo just remove him from navigation.

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Nice design!
See my comments below.

File: submissions/kovalov/html-css-popup/styles.css (hunk 15 above)
Comment (kovalov) [reply submissions/kovalov/html-css-popup/styles.css:236]: Have added it to hide the input and provide ability to click the items inside.

File: submissions/kovalov/html-css-popup/index.html [hunk 17]
```diff
@@ -13,94 +13,115 @@
   <body>
     <header class="header">
       <div class="header__container">
+        <a class="header__title" href="#"> John Doe </a>
```
Comment (A-Ostrovnyy) [inline submissions/kovalov/html-css-popup/index.html:16]: Logo should lead to the home page.

File: submissions/kovalov/html-css-popup/styles.css [hunk 18]
```diff
@@ -161,12 +175,12 @@ body {
 	appearance: none;
 	cursor: pointer;
 	outline: none;
-	border: 2px solid transparent;
+	/* border: 2px solid transparent; */
```
Comment (A-Ostrovnyy) [inline submissions/kovalov/html-css-popup/styles.css]: Avoid adding commented code to the commit.

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Nice!
Almost done, see two comments below.

File: submissions/kovalov/html-css-popup/index.html (hunk 17 above)
Comment (kovalov) [reply submissions/kovalov/html-css-popup/index.html]: Have added '/' (should use './'?)

File: submissions/kovalov/html-css-popup/styles.css (hunk 18 above)
Comment (kovalov) [reply submissions/kovalov/html-css-popup/styles.css]: Haven't found commented code line.

File: submissions/kovalov/html-css-popup/index.html (hunk 17 above)
Comment (kovalov) [reply submissions/kovalov/html-css-popup/index.html]: have added './' and it works charm ✨✨✨

**Review by A-Ostrovnyy — APPROVED** [review]
Well done!
